src/components/fuel/index.tsx

```diff
@@ -53,6 +57,13 @@ export default function FuelForm() {
   const submitFuelForm = async () => {
     setLoading(true);
     try {
+      const { Pin } = await getCaptain(data.Captain) as Captain
+      const CaptainPin = Pin.replace(/\n/g, "");
+
+      if (CaptainPin !== data.Pin) {
+        return toast.error(t("error.error_captain_pin"))
+      }
```

Comment:
This check needs to be done in the API, the PIN can not reach the client (this is a security concern)